Move airflow conf fetch call to setting.py (#975)
astronomer/astronomer-cosmos: docs/configuration/cosmos-conf.rst

Cosmos Config
=============

> This page contains the list of all available Airflow configurations and environment variables specific to ``astronomer-cosmos`` that can be set in the airflow.cfg file or using environment variables.

Reviewer: ```suggestion
This page lists all available Airflow configurations that affect ``astronomer-cosmos`` Astronomer Cosmos behavior. They can be set in the ``airflow.cfg file`` or using environment variables.
```

.. note::
    For more information, see `Setting Configuration Options <https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html>`_.

**Sections:**

- [cosmos]
- [openlineage]

> [cosmos]

Reviewer: Could you give an example of how the environment variable looks for one of these?

Author: I have added Default and Environment Variable for each config PTAL
<img width="760" alt="Screenshot 2024-05-20 at 4 19 01 PM" src="https://github.com/astronomer/astronomer-cosmos/assets/98807258/5450a229-9a40-44e6-a8a2-d1a75aeb33ee">

~~~~~~~~

1. ``cache_dir``:
    The directory used for caching Cosmos data.

    - Default: ``{TMPDIR}/cosmos_cache`` (where ``{TMPDIR}`` is the system temporary directory)

2. ``enable_cache``:
    Enable or disable caching of Cosmos data.

    - Default: ``True``

3. ``propagate_logs``:
    Whether to propagate logs in the Cosmos module.

    - Default: ``True``

4. ``dbt_docs_dir``:
    The directory path for dbt documentation.

    - Default: ``None``

5. ``dbt_docs_conn_id``:
    The connection ID for dbt documentation.

    - Default: ``None``

[openlineage]
~~~~~~~~~~~~~

> 

Reviewer: I think it's worth adding a reference that this is the OpenLineage provider configuration, and where users can 
https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/guides/user.html

Author: I have added a note on this.
